aries_cloudagent/protocols/out_of_band/v1_0/messages/invitation.py

```diff
@@ -140,12 +143,20 @@ def __init__(
         )
         self.requests_attach = list(requests_attach) if requests_attach else []
         self.services = services
+        self.assign_version_to_message_type(version=version)
 
     @classmethod
     def wrap_message(cls, message: dict) -> AttachDecorator:
         """Convert an aries message to an attachment decorator."""
         return AttachDecorator.data_json(mapping=message, ident="request-0")
 
+    @classmethod
+    def assign_version_to_message_type(cls, version: str):
+        """Assign version to Meta.message_type."""
+        cls.Meta.message_type = Template(cls.Meta.message_type).substitute(
```

Comment:
Wouldn't this overwrite message_type for all the class instances going forward?

Reply:
To update `message_type` in Meta, I found this can only be done in the classmethod before initialization. Regarding overwriting, `$version` is only present in `message_types` for OOB v1_0, so it will overwrite. But when we have a hardcoded version such as in other protocols then it won't overwrite.